### ISSUE TITLE
Ore Redemption Machine / UI

### ISSUE THREAD

**Issue by chairbender** [unitystation/unitystation#2257]
# Description
Depends on #2256 ,  #2301 , #2258 , #2458 

Implement the ore redemption machine and associated UI. You should be able to drag an ore box in front of it and they will automatically be loaded into the ORM. You should also be able to dump your satchel into the ORM by clicking the ORM with the satchel in the active hand or while activating the satchel while standing next to the ORM.

I don't think it needs to use inventory system for holding ores, it can just store an internal count of each type of ore.

When ore is added it adds unclaimed credits to the machine, which can be claimed by inserting an ID and selecting "claim". You'll need to add a way to store mining credits on the ID card (this can just be a server-side only miningCredits field on the IDCard component).

The UI itself allows you to enter a number of sheets to print of a given material, provided that you have sufficient ores for it. It also allows smelting sheets if you have the right combination of materials for them.

The different sheet materials which it can spit out should be separate prefab variants.

Additional info about ORM and other mining stuff can be found here: https://tgstation13.org/wiki/Shaft_Miner

You can see it in action here:
https://youtu.be/KOtfRZt6F3s?t=1148

TODO: Somebody with more knowledge fill in the details of how it should work

**Comment by Bod9001** [comment]
Would it be okay implementing a basic version of the ORM  that just spits out the  materials when you put the ores next to it

**Comment by PerfectTangent** [comment]
closing in favour of #6151
